core/good_key.go

```diff
@@ -63,9 +64,80 @@ func GoodKey(key crypto.PublicKey) error {
 // GoodKeyECDSA determines if an ECDSA pubkey meets our requirements
 func GoodKeyECDSA(key ecdsa.PublicKey) (err error) {
 	log := blog.GetAuditLogger()
-	err = NotSupportedError("ECDSA keys not yet supported")
-	log.Debug(err.Error())
-	return
+
+	// Check the curve.
+	err = GoodCurve(key.Curve)
+	if err != nil {
+		return err
+	}
+
+	// Assuming a prime field since a) we are only allowing such curves
+	// and b) crypto/elliptic only supports prime curves.
+	params := key.Params()
+
+	// Partial check of the public key for an invalid range in the EC group:
+	//   Verify that key is not the point at infinity O.
+	if isAdditiveIdentity(key.Curve, key.X, key.Y) {
+		err = MalformedRequestError("Key x, y must not be the point at infinity")
+		log.Debug(err.Error())
```

Comment:
The existing `log.Debug(err.Error())` in this function was a mistake - there's not enough context in the logged message, and errors returned from this function will get logged by the caller anyhow. Please remove these debug log statements. That should also allow you to move to a clearer `return MalformedRequestError(...)`, and remove the `log := blog.GetAuditLogger()` above.